nemo_curator/modules/fuzzy_dedup.py

```diff
@@ -362,21 +369,42 @@ def lsh(
                     )
                 df2.to_parquet(write_path, write_index=False, overwrite=True)
             else:
-                df2.to_parquet(write_path, write_index=False, append=True)
+                df2.to_parquet(
+                    write_path,
+                    write_index=False,
+                    overwrite=buckets_isempty,
+                    append=not buckets_isempty,
+                )
 
-            self._logger.info(f"Wrote data for buckets: {value_vars}")
+            if os.path.exists(write_path) and buckets_isempty:
+                buckets_isempty = check_empty_buckets(write_path)
+
+            if buckets_isempty:
+                self._logger.info(
+                    f"No duplicate documents found for buckets: {value_vars}"
+                )
+            else:
+                self._logger.info(f"Wrote data for buckets: {value_vars}")
+
+        if buckets_isempty:
+            self._logger.info("No duplicate documents found during LSH")
+            import shutil
+
+            shutil.rmtree(write_path)
```

Comment:
Agreed. Decided to go via this route for now (since other places also use shutil). Aligned that the refactor to be more remote friendly should leverage fsspec utilities where possible.